addon/mixins/data-adapter-mixin.js

```diff
@@ -30,10 +30,14 @@ export default Ember.Mixin.create({
   session: service('session'),
 
   ajaxOptions() {
+    const authorizer = this.get('authorizer');
     let hash = this._super(...arguments);
     let { beforeSend } = hash;
+
+    Ember.assert("You're using the DataAdapterMixin without specifying an authorizer. Please add `authorizer: 'authorizer:application'` to your adapter.", Ember.isPresent(authorizer));
+
```

Comment:
I think this should be done before anything else in this method, e.g. right after `const authorizer = this.get('authorizer');`.